contracts/shared/helpers/Fixture.ts

```diff
@@ -75,115 +40,89 @@ export default class Fixture {
     secretNumbers?: number[]
   ) {
     let chainId = (await ethers.provider.getNetwork()).chainId;
-    let provider = ethers.provider;
 
     let allSigners = await ethers.getSigners();
     let signers = (allSigners).slice(0, Fixture.ECDSA_ACCOUNTS_LENGTH);
     let addresses = await Promise.all(signers.map(acc => acc.getAddress())) as string[];
 
-    let blsSignerFactory = await BlsSignerFactory.new();
-    let blsSigners = new Array(blsWalletCount);
-    for (let i=0; i<blsSigners.length; i++) {
-      let secretNumber = Math.abs(Math.random() * 0xffffffff << 0);
-      if (secretNumbers) {
-        secretNumber = secretNumbers[i]; // error here if not enough numbers provided
-      }
-      blsSigners[i] = blsSignerFactory.getSigner(DOMAIN, "0x"+secretNumber.toString(16));
-    }
-
     let create2Fixture = Create2Fixture.create();
 
     // deploy wallet implementation contract
-    let blsWalletImpl = await create2Fixture.create2Contract("BLSWallet") as BLSWallet;
+    let blsWalletImpl = await create2Fixture.create2Contract("BLSWallet");
     try {
       await (await blsWalletImpl.initialize(
         ethers.constants.AddressZero
       )).wait();
     } catch (e) {}
 
     // deploy Verification Gateway
-    let verificationGateway = await create2Fixture.create2Contract("VerificationGateway") as VerificationGateway;
+    let vgContract = await create2Fixture.create2Contract("VerificationGateway");
     let bls = await create2Fixture.create2Contract("BLSOpen");
 
     try {
-      await (await verificationGateway.initialize(
+      await (await vgContract.initialize(
         bls.address,
         blsWalletImpl.address
       )).wait();
     } catch (e) {}
 
     // deploy BLSExpander Gateway
-    let blsExpander = await create2Fixture.create2Contract("BLSExpander") as BLSExpander;
+    let blsExpander = await create2Fixture.create2Contract("BLSExpander");
     try {
-      await (await blsExpander.initialize(verificationGateway.address)).wait();
+      await (await blsExpander.initialize(vgContract.address)).wait();
     } catch (e) {}
 
+    const verificationGateway = new VerificationGateway(vgContract.address, vgContract.signer);
```

Comment:
This naming will be less conventional since it is not the verification gateway contract (ethers/typechain), but rather a wrapper to interact with it.

Reply:
As discussed "VerificationGateway.ts" will be replaced with typechain type. And BLSWallet.ts can have a more appropriate name leaving BLSWallet as the typechain type.